lib/location.js

```diff
@@ -80,19 +161,25 @@ var LocationState = State.extend(function (self, name, opts) {
 
     // Defaults
     opts = _.defaults(opts || {}, {
-        next:null, 
+        next:null,
         question:'What is your address?',
         refine_question:'Please select your location from the following:',
         error_question:[
                         'Error: No results for your search term.',
                         'Please try another search term.'
                         ].join('\n'),
+        mapping_service:"gmaps",
         options_per_page:8,
         characters_per_page:160,
         next_text:"Next",
         previous_text:"Previous",
-        store_fields:['formatted_address'],
-        namespace:'location'
+        store_fields: ((opts.mapping_service === 'osmaps') ?
```

Comment:
If your other default is gmaps I think the default here should be that too. Don't try and get too smart IMHO.